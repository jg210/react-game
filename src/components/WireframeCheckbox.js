// @flow
//
// (c) 2018-2019 Jeremy Green

import React, { Component } from 'react';
import { connect } from "react-redux";

import { setWireframeMode } from '../redux/actions';
import type { DebugState } from '../redux/reducers/debug';

type Props = {
  enabled: boolean,
  setWireframeMode: (boolean) => void
}

export class WireframeCheckbox extends Component<Props> {

  render() {
    return (
      <div>
        <span>          
          <input type="checkbox" value={this.props.enabled} onChange={this.handleClick}/>
          debug
        </span>
      </div>
    );
  }

  handleClick = (event: SyntheticEvent<HTMLInputElement>) => {
    const target: HTMLInputElement = event.currentTarget;
    this.props.setWireframeMode(target.checked);
  }

}

const mapStateToProps = (state: {debug: DebugState}) => {
  const enabled = state.debug.wireframe;
  return { enabled };
};
const actionCreators = {
  setWireframeMode
};
export default connect(mapStateToProps, actionCreators)(WireframeCheckbox);
