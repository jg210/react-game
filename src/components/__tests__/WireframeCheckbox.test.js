// @flow
//
// (c) 2019 Jeremy Green

import React from 'react';
import { WireframeCheckbox } from '../WireframeCheckbox';

import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

it('renders correctly', () => {
  const tree = renderer
    .create(<WireframeCheckbox
      enabled={true}
      setWireframeMode={(enabled: boolean) => {}}
    />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('clicking', () => {
  const handler = jest.fn();
  const component = mount(<WireframeCheckbox
    enabled={false}
    setWireframeMode={handler}/>
  );
  const input = component.find('input');
  const domNode = input.getDOMNode();
  domNode.checked = true;
  input.simulate('change', { target: domNode });
  expect(handler).toHaveBeenCalledTimes(1);
  expect(handler).toHaveBeenCalledWith(true);
});