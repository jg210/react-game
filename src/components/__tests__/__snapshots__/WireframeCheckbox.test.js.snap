// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <span>
    <input
      onChange={[Function]}
      type="checkbox"
      value={true}
    />
    debug
  </span>
</div>
`;
